CMakeLists.txt

```diff
@@ -191,12 +192,12 @@ endif()
 # We want Kokkos to be built with C++14, since that's what we're using in
 # Parthenon.
 set(Kokkos_CXX_STANDARD 14)
-if(EXISTS ${Kokkos_ROOT}/CMakeLists.txt)
+if(EXISTS ${EXTERNAL_KOKKOS})
+  find_package(Kokkos 3 REQUIRED PATHS "${EXTERNAL_KOKKOS}" NO_DEFAULT_PATH)
```

Comment:
Why are you using NO_DEFAULT_PATH? 

I think you will be fine if you don't have the keyword NO_DEFAULT_PATH. Without this keyword the path provided with EXTERNAL_KOKKOS should be prioritized, if it is not found there it will look in the default locations? Is this not what you want? 

Are you wanting to completely avoid looking in default locations?

Reply:
My reading of [this documentation](https://cmake.org/cmake/help/v3.0/command/find_package.html) is that paths in `PATHS` are checked *last* of all possible paths unless other paths are disabled by `NO_DEFAULT_PATH`. Maybe I'm mistaken about that, I found that logic surprising.

Reply:
Ya, if that's true, it's just sad... Regardless you will probably still want to search the default PATHS as an option. For instance if an PATH has not been explicitly set perhaps favor an external kokkos installation over using the the submodule. 

```
find_package(Kokkos 3 REQUIRED) 

```

Reply:
Ok I've added `find_package(Kokkos 3 REQUIRED)` as a last chance if `EXTERNAL_KOKKOS` and `Kokkos_ROOT` have not been set and `external/Kokkos/CMakeLists.txt` does not exist. I think the submodule case should work if it exists, and users should have to explicitly bring it in/can get rid of it if they don't want it (or just use `EXTERNAL_KOKKOS` for their kokkos installation).

Reply:
Lol, you need to add it above the `add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/Kokkos Kokkos))` option, that option is always available so really in it's current order I don't think `find_package(Kokkos 3 REQUIRED)` will ever get triggered. Sorry i wasn't more clear. The only reason /external/Kokkos/CMakeLists.txt would not be available is if you did not do a git clone --recursive.

Reply:
You want `HINTS`, not `PATHS` to prioritize the path.

Reply:
Definitely don't do `NO_DEFAULT_PATH`.

Reply:
But, anyway, `Kokkos_ROOT` is the built-in way to do this. Writing a separate comment to force usage of an external kokkos or subdirectory kokkos.